### PR TITLE
Refactored fields and broke it to ReadFields and UpdateFields

### PR DESCRIPTION
**Pull request by JigarJoshi**

### CODE REVIEW

File: client/src/main/java/com/tigrisdata/db/client/model/UpdateFields.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  }
+
+  public static class SetFieldsBuilder {
+    private final Map<String, Object> internalMap;
+
+    private SetFieldsBuilder() {
+      this.internalMap = new LinkedHashMap<>();
+    }
+
+    public SetFieldsBuilder set(String fieldName, int newValue) {
```
Comment (himank) [inline client/src/main/java/com/tigrisdata/db/client/model/UpdateFields.java:75]: I am assuming you are planning to add more setters for bytes etc?

File: client/src/main/java/com/tigrisdata/db/client/model/UpdateFields.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.tigrisdata.db.client.utils.Utilities;
+
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public final class UpdateFields {
+  private final Map<String, Object> internalMap;
+  private static final UpdateFields EMPTY = new UpdateFields(Collections.emptyMap());
```
Comment (himank) [inline client/src/main/java/com/tigrisdata/db/client/model/UpdateFields.java:12]: not needed?

File: client/src/main/java/com/tigrisdata/db/client/model/ReadFields.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.tigrisdata.db.client.utils.Utilities;
+
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public final class ReadFields {
+  private final Map<String, Object> internalMap;
+  private static final ReadFields EMPTY = new ReadFields(Collections.emptyMap());
```
Comment (himank) [inline client/src/main/java/com/tigrisdata/db/client/model/ReadFields.java:12]: empty means everything returned from the server even if fields is not in JSON.

File: client/src/test/java/com/tigrisdata/db/client/model/UpdateFieldsTest.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.nio.charset.StandardCharsets;
+
+public class UpdateFieldsTest {
+
+  @Test
+  public void setFields() throws JsonProcessingException {
```
Comment (himank) [inline client/src/test/java/com/tigrisdata/db/client/model/UpdateFieldsTest.java:12]: nit, can you add test when fields are not set what is the json looks like?

File: client/src/test/java/com/tigrisdata/db/client/model/ReadFieldsTest.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
+package com.tigrisdata.db.client.model;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class ReadFieldsTest {
```
Comment (himank) [inline client/src/test/java/com/tigrisdata/db/client/model/ReadFieldsTest.java:7]: nit, can you add test what should be done to include all?

**Review by himank — APPROVED** [review]

File: client/src/main/java/com/tigrisdata/db/client/model/UpdateFields.java (hunk 1 above)
Comment (JigarJoshi) [reply client/src/main/java/com/tigrisdata/db/client/model/UpdateFields.java:66]: discussed offline and added `byte[]`

File: client/src/main/java/com/tigrisdata/db/client/model/ReadFields.java (hunk 3 above)
Comment (JigarJoshi) [reply client/src/main/java/com/tigrisdata/db/client/model/ReadFields.java:12]: Client will now not send this field to server if it is empty.

File: client/src/test/java/com/tigrisdata/db/client/model/UpdateFieldsTest.java (hunk 4 above)
Comment (JigarJoshi) [reply client/src/test/java/com/tigrisdata/db/client/model/UpdateFieldsTest.java:17]: added test, if no set fields are set and user is trying to build, they will face an error.

File: client/src/test/java/com/tigrisdata/db/client/model/ReadFieldsTest.java (hunk 5 above)
Comment (JigarJoshi) [reply client/src/test/java/com/tigrisdata/db/client/model/ReadFieldsTest.java:7]: added test.

File: client/src/main/java/com/tigrisdata/db/client/model/UpdateFields.java (hunk 2 above)
Comment (JigarJoshi) [reply client/src/main/java/com/tigrisdata/db/client/model/UpdateFields.java]: removed.

**Pull request merged**